httpbp/response.go

```diff
@@ -0,0 +1,210 @@
+package httpbp
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"html/template"
+	"net/http"
+
+	httpgk "github.com/go-kit/kit/transport/http"
+)
+
+// HTTPError is a specialized error that is returned be the Err method specified
+// in the ErrorResponse interface.
+type HTTPError struct {
+	// Code is the status code to set on the HTTP response.  Defaults to 500 if
+	// it is not set.
+	Code int
+
+	// ResponseMessage is an optional message that can be returned to the
+	// client.  Defaults to the native http.StatusText message for the
+	// StatusCode() of the HTTPError.
+	ResponseMessage string
+
+	// Cause is an optional error that can be used to
+	Cause error
+}
+
+// Error returns the error message to send to the client.
+func (e HTTPError) Error() string {
+	if e.ResponseMessage != "" {
+		return e.ResponseMessage
+	}
+	return http.StatusText(e.StatusCode())
+}
```

Comment:
I suggest move this logic into a different function (maybe call it `ResponseMessage`), and for `Error` do something more standard:

```
func (e HTTPError) Error() string {
  return fmt.Sprintf("httpbp: http error with code %d, message %q and cause %v", e.Code, e.Message, e.Cause)
}
```